lib/qless/job.rb

```diff
@@ -110,6 +111,18 @@ def self.build(client, klass, attributes = {})
       new(client, attributes)
     end
 
+    # Converts a hash of job options (as returned by job.to_hash) into the array
+    # format the qless api expects.
+    def self.build_opts_array(opts)
+      result = []
```

Comment:
Maybe it'd be worth DRYing this (this assumes that JSON.generate will handle primitives properly).

```ruby
DEFAULT_OPTS = {
  'priority' => 0,
  'tags', => [],
  # etc
}

def self.build_opts_array(opts)
  DEFAULT_OPTS.reduce([opts.fetch(:delay, 0)]) do |result, (key, default)|
    value = JSON.generate(opts.fetch(key.to_sym, default))
    result << [key, value]
  end
end
```

It'd be great if you could just slice out what you're interested in from the `opts` Hash and apply defaults for the missing values.